routers/api/v1/repo/branch.go

```diff
@@ -300,6 +300,94 @@ func ListBranches(ctx *context.APIContext) {
 	ctx.JSON(http.StatusOK, apiBranches)
 }
 
+// RenameBranch rename a branch for a user's repository
+func RenameBranch(ctx *context.APIContext) {
+	// swagger:operation POST /repos/{owner}/{repo}/branches/{name}/rename repository repoRenameBranch
```

Comment:
what if the branch is named "test/branch/rename"? Won't the http router look for the branched named "test/branch"?

Reply:
when original branch is named `test/branch/rename`, the `ctx.params()` will return `test/branch/rename` too
![image](https://user-images.githubusercontent.com/33891828/220562282-cde6ce9d-2c2b-42c6-99fe-770cb28b9281.png)